src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.cs

```diff
@@ -470,45 +456,33 @@ private static int BinarySearch(string concatenatedNames, ImmutableArray<Node> n
 
             if (_nodes[node.ParentIndex].IsRoot)
             {
-                results.AddRange(rootContainer.GetMembers(GetName(node)));
+                var members = rootContainer.GetMembers(node.Name);
+                results.AddRange(members);
             }
             else
             {
-                using var _ = ArrayBuilder<ISymbol>.GetInstance(out var containerSymbols);
-
-                Bind(node.ParentIndex, rootContainer, containerSymbols, cancellationToken);
-
+                using var containerSymbols = TemporaryArray<ISymbol>.Empty;
+                Bind(node.ParentIndex, rootContainer, ref Unsafe.AsRef(in containerSymbols), cancellationToken);
+#pragma warning disable RS0042 // Do not copy value
                 foreach (var containerSymbol in containerSymbols)
+#pragma warning restore RS0042 // Do not copy value
```

Comment:
📝 RS0042 doesn't know how to handle `foreach` usage right now, but I'll update it and the suppressions related to this usage will go away.

Reply:
➡️ RS0042 has been updated to handle this